src/components/organisms/AssetActions/Consume.tsx

```diff
@@ -124,35 +130,60 @@ export default function Consume({
   }, [dtBalance])
 
   useEffect(() => {
-    setIsDisabled(
-      (!ocean ||
-        !isBalanceSufficient ||
-        typeof consumeStepText !== 'undefined' ||
-        pricingIsLoading ||
-        !isConsumable) &&
-        !hasPreviousOrder &&
-        !hasDatatoken
-    )
+    async function validateAsset() {
+      const did = DID.parse(ddo.id)
+      const fileValid = await isFileValid(
+        did,
+        ddo.findServiceByType('access').serviceEndpoint,
+        axios.CancelToken.source().token
```

Comment:
Not the proper way to handle cancel token, check here https://github.com/oceanprotocol/market/blob/1da7eae35a1bfe24bced84d99089638b772990b5/src/providers/Asset.tsx#L162